src/walletConnect/index.tsx

```diff
@@ -383,10 +393,7 @@ export async function onSessionProposal(
   const supportedChainIds = chainIds.filter(isSupportedChain);
   const unsupportedChainIds = chainIds.filter(id => !isSupportedChain(id));
 
-  if (
-    (unsupportedChainIds.length && !supportedChainIds.length) ||
-    unsupportedNamespaces.length
-  ) {
+  if (unsupportedChainIds.length) {
```

Comment:
This was a mistake in my logic that `optionalNamespaces` and the builder util clarify. A few lines up we reject if any non eip155 namespaces were required. Here, we reject if any unsupported chains are required.